where-to-travel/src/main/webapp/script.js

```diff
@@ -1,141 +1,302 @@
     mapTypeId: google.maps.MapTypeId.ROADMAP,
     zoom: 16,
     mapTypeControl: false,
-    styles: mapStyles
-  };	 
-  map = new google.maps.Map(document.getElementById("map"), mapOptions);
+    styles: mapStyles,
+  };
+  map = new google.maps.Map(document.getElementById('map'), mapOptions);
 
   let homeMarker = new google.maps.Marker({
     position: home,
+    icon: 'icons/home.svg',
     map: map,
-    title: 'Home'
+    title: 'Home',
   });
+  map.addListener('click', toggleFocusOff);
+}
+
+function attachFormValidators() {}
+
+function addLoginButtons() {
+  let dashElement = $(getLoginHtml());
+  $('#' + dashId).append(dashElement);
+}
+
+function addUserDash() {
+  let dashElement = $(getUserDashHtml(user));
+  $('#' + dashId).append(dashElement);
+}
+
+/** Toggle the focussed pin/card off */
+function toggleFocusOff() {
+  if (focussedCard != null) {
+      focussedCard.classList.remove('active');
+    }
+
+    if (focussedPin != null) {
+      focussedPin.setIcon('icons/pin.svg');
+    }
+    focussedCard = null;
+    focussedPin = null;
 }
 
 /**
- * Responds to click on submit button by getting places and placing pins.
+ * Responds to click on submit button by getting input time from user,
+ * finding places within requested time, and placing corresponding pins
+ * on the map .
  *
- * @param {Event} event click event to respond to
+ * @param {Event} event Click event from which to respond
  */
 function submitDataListener(event) {
+  clearPlaces();
   const hours = document.getElementById(hoursId).value;
   const minutes = document.getElementById(minutesId).value;
-  const timeObj = { "hours": hours, "minutes": minutes };
-  getPlacesFromTime(timeObj).then(places => populatePlaces(places));
+  // Convert hours and minutes into seconds
+  const time = hours * 3600 + minutes * 60;
+  getPlacesFromTime(time).then(places => {
+    populatePlaces(places);
+  });
 }
 
 /**
- * Populates map with pins. Given a list of places, put markers at each
- * lat/long location.
+ * Populates display with places returned by a query. First, adds pins to the map. Then, adds
+ * cards to the location card scroller in the DOM.
  *
- * @param {array} placeArray array of places to place markers at
+ * @param {array} placeArray Array of Google Maps Place Objects
  */
 function populatePlaces(placeArray) {
-  let i;
-  for(i = 0; i < placeArray.length; i++) {
+  for(let i = 0; i < placeArray.length; i++) {
     let name = placeArray[i].name;
     let coordinates = placeArray[i].geometry.location;
+
+    // TODO: Use this link to provide directions to user
+    let directionsLink = 'https://www.google.com/maps/dir/' +
+      home.lat + ',' + home.lng + '/' +
+      coordinates.lat() + ',' + coordinates.lng();
+
+    console.log(directionsLink);
+    let timeStr = placeArray[i].timeAsString;
+
     let placeMarker = new google.maps.Marker({
       position: coordinates,
       map: map,
-      title: name
+      title: name,
+      icon: 'icons/pin.svg',
```

Comment:
maybe make 'icons/pin.svg' a  global constant like ICON_PATH since it seems to be repeated a couple of times.

Reply:
fixed in 54ef408